jmespath-core/src/main/java/io/burt/jmespath/BaseAdapter.java

```diff
@@ -60,8 +58,8 @@ public int compare(T value1, T value2) {
         case BOOLEAN:
           return (isTruthy(value1) && isTruthy(value2)) || (!isTruthy(value1) && !isTruthy(value2)) ? 0 : -1;
         case NUMBER:
-          Comparable d1 = toNumber(value1).doubleValue();
-          Comparable d2 = toNumber(value2).doubleValue();
+          Double d1 = toNumber(value1).doubleValue();
+          Double d2 = toNumber(value2).doubleValue();
           return d1.compareTo(d2);
```

Comment:
Shouldn't this be `double d1` and `Double.compare(d1, d2)` to avoid boxing?

While not related to this PR, the logic for Boolean should probably be changed in a similar way, as there seems to be no way of returning a positive number as is (and the documentation lists Boolean amongst the comparable values).